archive/miu_language/arithmetic.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2020 Gihan Marasingha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gihan Marasingha
+-/
+import data.nat.modeq
+import tactic.linarith
+
+/-!
+# Arithmetic
+
+A basic arithmetic result needed to analyse the MIU system.
+-/
+
+open nat
+
+/--
+If `c ≡ 1 [MOD 3]`, then there exists `k`, a power of 2 for which `c ≡ k [MOD 3]` and `c ≤ k`.
+-/
+private lemma mod1pow (x : ℕ) : ∃ m : ℕ, 1 + 3*x ≤ 2^m ∧ 2^m ≡ 1 [MOD 3] :=
+begin
+  induction x with k hk, { 
+    use 2, /- base case -/
+    split;
+      norm_num,
+    refl,
+  }, { /- Induction step -/
```

Comment:
I think we're going to want archive material to be an exemplar of "idiomatic Lean", including following the style guide.

Could you have a look at https://leanprover-community.github.io/contribute/style.html and try to follow it obsessively? The thing I noticed here was "braces on their own line".

Reply:
I've updated the code. I _think_ it now adheres to the style guide. I take it that the variable naming convention is not inflexible - certainly the guide itself doesn't follow the variable naming convention.

Do let me know if you spot anything amiss. Thanks!

Reply:
One other question: by refactoring, I've considerably reduced the size of `arithmetic.lean`. I now think I should remove this file and insert the lemmas from `arithmetic.lean` into the `decision_suf.lean` file where needed. Do you agree?